data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/GetAttestationDataTest.java

```diff
@@ -13,120 +13,67 @@
-  }
+    when(validatorDataProvider.createAttestationDataAtSlot(UInt64.ONE, 1))
+        .thenReturn(SafeFuture.completedFuture(Optional.of(attestationData)));
 
-  @Test
-  void shouldRejectTooFewArguments() throws Exception {
-    badRequestParamsTest(Map.of(), "Please specify both slot and committee_index");
-  }
+    handler.handleRequest(request);
 
-  @Test
-  void shouldRejectWithoutSlot() throws Exception {
-    badRequestParamsTest(
-        Map.of("foo", List.of(), "Foo2", List.of()), "'slot' cannot be null or empty.");
+    assertThat(request.getResponseCode()).isEqualTo(SC_OK);
+    assertThat(request.getResponseBody()).isEqualTo(attestationData);
   }
 
   @Test
-  void shouldRejectWithoutCommitteeIndex() throws Exception {
-    badRequestParamsTest(
-        Map.of(SLOT, List.of("1"), "Foo2", List.of()),
-        "'committee_index' cannot be null or empty.");
+  void metadata_shouldHandle400() throws JsonProcessingException {
+    verifyMetadataErrorResponse(handler, HttpStatusCodes.SC_BAD_REQUEST);
   }
 
   @Test
-  void shouldRejectNegativeCommitteeIndex() throws Exception {
-    badRequestParamsTest(
-        Map.of(SLOT, List.of("1"), COMMITTEE_INDEX, List.of("-1")),
-        "'committee_index' needs to be greater than or equal to 0.");
+  void metadata_shouldHandle404() throws JsonProcessingException {
+    verifyMetadataErrorResponse(handler, HttpStatusCodes.SC_NOT_FOUND);
   }
 
   @Test
-  void shouldReturnNoContentIfNotReady() throws Exception {
-    Map<String, List<String>> params = Map.of(SLOT, List.of("1"), COMMITTEE_INDEX, List.of("1"));
-
-    when(context.queryParamMap()).thenReturn(params);
-    when(provider.createAttestationDataAtSlot(UInt64.ONE, 1))
-        .thenReturn(SafeFuture.failedFuture(new ChainDataUnavailableException()));
-    handler.handle(context);
-
-    verify(context).future(resultCaptor.capture());
-    final SafeFuture<String> result = resultCaptor.getValue();
-    assertThat(result).isCompletedExceptionally();
-    assertThatThrownBy(result::join).hasRootCauseInstanceOf(ChainDataUnavailableException.class);
+  void metadata_shouldHandle500() throws JsonProcessingException {
+    verifyMetadataErrorResponse(handler, SC_INTERNAL_SERVER_ERROR);
   }
 
   @Test
-  void shouldReturnAttestationData() throws Exception {
-    Map<String, List<String>> params = Map.of(SLOT, List.of("1"), COMMITTEE_INDEX, List.of("1"));
-
-    when(context.queryParamMap()).thenReturn(params);
-    when(provider.isStoreAvailable()).thenReturn(true);
-    when(provider.createAttestationDataAtSlot(UInt64.ONE, 1))
-        .thenReturn(SafeFuture.completedFuture(Optional.of(attestationData)));
-    handler.handle(context);
-
-    verify(context).future(resultCaptor.capture());
-    final SafeFuture<String> result = resultCaptor.getValue();
-    assertThat(result)
-        .isCompletedWithValue(
-            jsonProvider.objectToJSON(new GetAttestationDataResponse(attestationData)));
-  }
```

Comment:
we'd need to tweak results, but this test should be ok too?

Reply:
That test is already covered in the new test cases.